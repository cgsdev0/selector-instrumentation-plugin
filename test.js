const pluginTester = require("babel-plugin-tester");
const selectorInstrumentationPlugin = require("../selector-instrumentation-plugin");

pluginTester({
  plugin: selectorInstrumentationPlugin,
  pluginName: "selectorInstrumentationPlugin",
  title: "selectorInstrumentationPlugin",
  tests: [
    {
      code: `export const selectNothing = createSelector();`,
      snapshot: true,
    },
    {
        code: `export const selectState = createSelector(state, state => state.get("stuff"));`,
      snapshot: true,
    },
    {
        code: `import { defaultMemoize } from "reselect";`,
        snapshot: false,
    },
    {
        code: `import { createSelector as reselectCreateSelector } from "reselect";`,
        snapshot: false,
    },
    {
        code: `import "reselect";`,
        snapshot: false,
    },
    {
        code: `import { createSelector } from "reselect";`,
        snapshot: true,
    },
    {
        code: `import { defaultMemoize, createSelector } from "reselect";`,
        snapshot: true,
    },
    {
        code: `export const selectFromMultipleThings = createSelector(selectFrom1, selectFrom2, selectFrom3, (t1, t2, t3) => t1 + t2 / t3);`,
        snapshot: true,
    },
  ],
});
