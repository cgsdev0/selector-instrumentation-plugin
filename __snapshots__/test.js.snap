// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`selectorInstrumentationPlugin 1. selectorInstrumentationPlugin: 1. selectorInstrumentationPlugin 1`] = `
"
export const selectNothing = createSelector();

      ↓ ↓ ↓ ↓ ↓ ↓

export const selectNothing = makeCreateSelector(\\"selectNothing\\")();
"
`;

exports[`selectorInstrumentationPlugin 2. selectorInstrumentationPlugin: 2. selectorInstrumentationPlugin 1`] = `
"
export const selectState = createSelector(state, state => state.get(\\"stuff\\"));

      ↓ ↓ ↓ ↓ ↓ ↓

export const selectState = makeCreateSelector(\\"selectState\\")(state, state => state.get(\\"stuff\\"));
"
`;

exports[`selectorInstrumentationPlugin 6. selectorInstrumentationPlugin: 6. selectorInstrumentationPlugin 1`] = `
"
import { createSelector } from \\"reselect\\";

      ↓ ↓ ↓ ↓ ↓ ↓

import { makeSelectorCreator } from \\"make-create-selector\\";
"
`;

exports[`selectorInstrumentationPlugin 7. selectorInstrumentationPlugin: 7. selectorInstrumentationPlugin 1`] = `
"
import { defaultMemoize, createSelector } from \\"reselect\\";

      ↓ ↓ ↓ ↓ ↓ ↓

import { defaultMemoize } from \\"reselect\\";
import { makeSelectorCreator } from \\"make-create-selector\\";
"
`;

exports[`selectorInstrumentationPlugin 8. selectorInstrumentationPlugin: 8. selectorInstrumentationPlugin 1`] = `
"
export const selectFromMultipleThings = createSelector(selectFrom1, selectFrom2, selectFrom3, (t1, t2, t3) => t1 + t2 / t3);

      ↓ ↓ ↓ ↓ ↓ ↓

export const selectFromMultipleThings = makeCreateSelector(\\"selectFromMultipleThings\\")(selectFrom1, selectFrom2, selectFrom3, (t1, t2, t3) => t1 + t2 / t3);
"
`;
